### PR TITLE
[5.x] Yarn v3 for stable webpacker 5.x gem

### PR DESCRIPTION
**Pull request by pedrofurtado**
Yarn v3 is released! Webpacker is already ready for this 🎉

### CODE REVIEW

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -29,14 +29,6 @@ jobs:
             gemfile: gemfiles/Gemfile-rails.6.0.x
         experimental: [false]
         include:
-          - ruby: 2.5
```
Comment (guillaumebriday) [inline .github/workflows/ruby.yml:32]: Why did you remove these versions?

Comment (pedrofurtado) [reply .github/workflows/ruby.yml:32]: @guillaumebriday Because Rails 7.0.0-alpha does not support ruby 2.5 and 2.6, only 2.7+ 👍

**Comment by imajes** [comment]
any chance we can get this merged and released soon? we rely on a yarn package that needs yarn 3, and so webpacker is continually complaining :(

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (rafaeldev) [reply .github/workflows/ruby.yml:32]: I think so these removes can be done in other PR separately, only that change context to rails 7. What do you think about that?

Comment (guillaumebriday) [reply .github/workflows/ruby.yml:32]: Yes I think it's should not be in this PR

Comment (pedrofurtado) [reply .github/workflows/ruby.yml:32]: Considering that branch `5-x-stable` does not receive frequent changes, and that this PR is small enough to be understood well (despite CI fixes be together with adding of Yarn v3), can we merge it? 🍻

**Comment by pedrofurtado** [comment]
@imajes For a while, in order to not block your team on Yarn upgrading, please use this fork:

```ruby
# Gemfile
gem 'webpacker', github: 'pedrofurtado/webpacker', branch: 'v5.4.0-with-yarnv3'

# package.json
{
  "dependencies": {
    "@pedrofurtado/webpacker": "latest"
  }
}

We, here, are using this fork, while a new official 5.x release is not rolled out 🤝 
```

**Comment by imajes** [comment]
@pedrofurtado i have already been using your branch, alas. I was hoping for something a bit more official... especially from this team :/

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (imajes) [reply .github/workflows/ruby.yml:32]: @pedrofurtado sounds like the preferred way forward is for you to pull the workflow changes out of this PR, focus it 💯 on just bringing yarn v3 support in for a quick merge....

Comment (pedrofurtado) [reply .github/workflows/ruby.yml]: PR splitted in two: this that provides yarn 3 support, and other that fixes CI ( #3106 )
